Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -198,12 +202,29 @@ public struct PackageBuilder {
         guard let pkgConfig = manifest.package.pkgConfig else { return nil }
         return RelativePath(pkgConfig)
     }
+
+    // FIXME: Should this be an extension on FileSystem and placed in Utility?
+    /// Returns path to all the items in a directory.
+    func directoryContents(_ path: AbsolutePath) throws -> [AbsolutePath] {
+        return try fileSystem.getDirectoryContents(path).map { path.appending(component: $0) }
+    }
+
+    // FIXME: Optimize to probably a lazy iterator and find correct home for this method.
+    /// Returns path to all the items in a directory tree.
+    func recursiveDirectoryContents(_ path: AbsolutePath, predicate: @noescape (AbsolutePath) -> Bool) throws -> [AbsolutePath] {
+        return try directoryContents(path).flatMap { (entry) -> [AbsolutePath] in
+            if predicate(entry) {
+                return try recursiveDirectoryContents(entry, predicate: predicate)
+            }
+            return [entry]
+        }
+    }
     
     func sourceRoot() throws -> AbsolutePath {
-        let viableRoots = walk(packagePath, recursively: false).filter { entry in
+        let viableRoots = try directoryContents(packagePath).filter { entry in
             switch entry.basename.lowercased() {
```

Comment:
This is an example of what I mentioned before, we get the combined paths then suck out the `basename`.